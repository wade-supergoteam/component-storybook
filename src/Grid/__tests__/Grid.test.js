import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import { Grid, GridItem } from "..";

const debugDefault = true;

test("Create Default Grid", () => {
  const wrapper = renderer.create(<Grid />);
  expect(wrapper).toMatchSnapshot();
});

test("Create Default Grid with default theme", () => {
  const wrapper = renderer.create(<Grid defaulttheme />);
  expect(wrapper).toMatchSnapshot();
});

test("Create Default Grid with getGridVerticalRules", () => {
  const wrapper = renderer.create(<Grid getGridVerticalRules />);
  expect(wrapper).toMatchSnapshot();
});

test("Grid bottom", () => {
  const wrapper = renderer.create(<Grid bottom />);
  expect(wrapper).toMatchSnapshot();
});

test("Grid with props", () => {
  const wrapper = renderer.create(<Grid full defaultGutter rev />);
  expect(wrapper).toMatchSnapshot();
});

test("Middle Center Grid", () => {
  const wrapper = renderer.create(<Grid middle center />);
  expect(wrapper).toMatchSnapshot();
});

test("Grid item debug", () => {
  const wrapper = renderer.create(<GridItem debug={("Debug", debugDefault)} />);
  expect(wrapper);
});

test("Grid item", () => {
  const wrapper = renderer.create(<GridItem media={{ phone: 1, tablet: 1 }} />);
  expect(wrapper).toMatchSnapshot();
});

test("Grid item 2", () => {
  const wrapper = renderer.create(<GridItem media={{ phone: 1, tablet: 1 / 2 }} col={1 / 4} />);
  expect(wrapper).toMatchSnapshot();
});
