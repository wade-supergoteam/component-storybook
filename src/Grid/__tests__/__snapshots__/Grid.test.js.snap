// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Default Grid 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

<div
  className="c0"
/>
`;

exports[`Create Default Grid with default theme 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

<div
  className="c0"
/>
`;

exports[`Create Default Grid with getGridVerticalRules 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

<div
  className="c0"
/>
`;

exports[`Grid bottom 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

.c0 > * {
  vertical-align: bottom;
}

<div
  className="c0"
/>
`;

exports[`Grid bottom 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

.c0 > * {
  vertical-align: bottom;
}

<div
  className="c0"
/>
`;

exports[`Grid item 1`] = `
.c0 {
  width: NaN%;
}

@media (max-width:48em) {
  .c0 {
    width: 100%;
  }
}

@media (max-width:72.5em) and (min-width:48em) {
  .c0 {
    width: 100%;
  }
}

<div
  className="c0"
  media={
    Object {
      "phone": 1,
      "tablet": 1,
    }
  }
/>
`;

exports[`Grid item 2 1`] = `
.c0 {
  width: 25%;
}

@media (max-width:48em) {
  .c0 {
    width: 100%;
  }
}

@media (max-width:72.5em) and (min-width:48em) {
  .c0 {
    width: 50%;
  }
}

<div
  className="c0"
  media={
    Object {
      "phone": 1,
      "tablet": 0.5,
    }
  }
/>
`;

exports[`Grid item debug 1`] = `
.c0 {
  background-color: rgba(192,192,192,0.17775145431929396);
  width: NaN%;
}

<div
  className="c0"
/>
`;

exports[`Grid with props 1`] = `
.c0 {
  font-size: 0;
  margin-left: 0px;
  direction: rtl;
  text-align: left;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 0px;
}

.c0 > * {
  direction: ltr;
  text-align: left;
}

<div
  className="c0"
/>
`;

exports[`Middle Center Grid 1`] = `
.c0 {
  font-size: 0;
  margin-left: -20px;
  text-align: center;
}

.c0 > * {
  font-size: 16px;
  display: inline-block;
  vertical-align: top;
  width: 100%;
}

.c0 > * {
  padding-left: 20px;
}

.c0 > * {
  vertical-align: middle;
}

<div
  className="c0"
/>
`;
